components/collector/tests/source_collectors/junit/test_tests.py

```diff
@@ -45,3 +45,15 @@ async def test_empty_test_suites(self):
         """Test that a JUnit XML file with an empty testsuites node works."""
         response = await self.collect(get_request_text=self.JUNIT_XML_EMPTY_TEST_SUITES)
         self.assert_measurement(response, value="0", total="0")
+
+    async def test_repeated_test_case_names(self):
+        """Test that repeated test case names are distinguished."""
+        junit_xml_repeated_test_case_name = """<testsuites>
+        <testsuite timestamp="2009-12-19T17:58:59">
+            <testcase name="tc1" classname="cn1"/>
```

Comment:
Is it not possible for JUnit XML files to contain the same combination of test name + class name across files?
E.g. `file1`: `name=1 class=1` and  `file2` having the same entry: `name=1 class=1` ?

Reply:
I don't know. There's no official XSD for JUnit XML. Just a bunch of conventions. In the example data I received for this bug report test cases only had two attributes: `classname` and `name`, so there's not much we can do if the combination of those two would not be unique.